src/validation.cpp

```diff
@@ -176,11 +176,67 @@ bool CheckFinalTxAtTip(const CBlockIndex& active_chain_tip, const CTransaction&
     return IsFinalTx(tx, nBlockHeight, nBlockTime);
 }
 
+std::optional<LockPoints> CalculateLockPointsAtTip(CBlockIndex* tip,
+                                                   const CCoinsView& coins_view,
+                                                   const CTransaction& tx)
+{
+    assert(tip);
```

Comment:
Would it make sense to pass a `CBlockIndex&` instead and avoid checking for nullptr here? If this fn requires a non-nullptr `CBlockIndex`, I think the fn signature should reflect that and the callsite to take responsibility.

Reply:
Going to leave it as this to keep this PR in more reviewable state.